src/django/api/facility_history.py

```diff
@@ -0,0 +1,126 @@
+from api.constants import ProcessingAction
+from api.models import FacilityMatch
+
+
+def create_location_diff_from_history_entry(entry):
+    return {
+        'old': {
+            'type': 'Point',
+            'coordinates': [
+                entry.prev_record.location.x,
+                entry.prev_record.location.y,
+            ],
+        },
+        'new': {
+            'type': 'Point',
+            'coordinates': [
+                entry.location.x,
+                entry.location.y,
+            ],
+        },
+    }
+
+
+def create_facility_history_dictionary(entry):
+    updated_at = str(entry.history_date)
+    change_reason = entry.history_change_reason or ''
+    history_type_display = entry.get_history_type_display()
+
+    if 'FacilityLocation' in change_reason:
+        return {
+            'updated_at': updated_at,
+            'action': 'UPDATE',
+            'changes': {
+                'location': create_location_diff_from_history_entry(entry),
+            },
+        }
+    elif 'Merged with' in change_reason:
+        return {
+            'updated_at': updated_at,
+            'action': 'MERGE',
+            'detail': change_reason,
+        }
+    elif 'Promoted' in change_reason:
+        return {
+            'updated_at': updated_at,
+            'action': 'UPDATE',
+            'changes': {
+                'location': create_location_diff_from_history_entry(entry),
+                'name': {
+                    'old': entry.prev_record.name,
+                    'new': entry.name,
+                },
+                'address': {
+                    'old': entry.prev_record.address,
+                    'new': entry.address,
+                },
+            },
+        }
```

Comment:
Consider including the `detail` key with the content of `change_reason` to provide the API consumer with some more detailed context.

Reply:
I set these to be either change_reason or history_type_display, depending on whether a change_reason existed.